homeassistant/components/mqtt/mixins.py

```diff
@@ -744,7 +744,12 @@ async def async_discovery_update(
                 self.log_name,
                 discovery_hash,
             )
-            await self.async_update(discovery_payload)
+            try:
+                await self.async_update(discovery_payload)
+            except vol.MultipleInvalid as schema_exception:
+                send_discovery_done(self.hass, self._discovery_data)
```

Comment:
Instead of calling this from yet another place, maybe to try..finally?

Reply:
Done, the other change, I had to remove an additional line as it calls always now.